### PR TITLE
Revendor gardener@v1.7.0 and add restore and migrate operations

### PR DESCRIPTION
**Pull request by plkokanov**
**What this PR does / why we need it**:
Vendor gardener version 1.6.0 and add and implement restore and migrate operations to the actuator.

**Which issue(s) this PR fixes**:
Part of [#1631](https://github.com/gardener/gardener/issues/1631)

**Special notes for your reviewer**:
None

**Release note**:

```improvement operator
Vendored gardener v1.7.0.
Added implementation for `Restore` and `Migrate` operations.
```

### CODE REVIEW

File: pkg/healthcheck/add.go [hunk 1]
```diff
@@ -44,7 +44,8 @@ func RegisterHealthChecks(mgr manager.Manager, opts healthcheck.DefaultAddArgs)
 	return healthcheck.DefaultRegistration(
 		cilium.Type,
 		extensionsv1alpha1.SchemeGroupVersion.WithKind(extensionsv1alpha1.NetworkResource),
-		func() runtime.Object { return &extensionsv1alpha1.Network{} },
+		func() runtime.Object { return &extensionsv1alpha1.NetworkList{} },
+		func() extensionsv1alpha1.Object { return &extensionsv1alpha1.Network{} },
```
Comment (zanetworker) [inline pkg/healthcheck/add.go:48]: why is this changed to `extensionsv1alpha1.Object` ?

Comment (plkokanov) [reply pkg/healthcheck/add.go:48]: `type GetExtensionObjectFunc = func() extensionsv1alpha1.Object` 
doesn't accept a return type of runtime.Object (errors during compilation)

**Comment by plkokanov** [comment]
Want to test out the entire migration just in case, but I'll do it a bit later today if that's ok.

**Comment by zanetworker** [comment]
Please take your time.

File: go.mod [hunk 2]
```diff
@@ -4,23 +4,22 @@ go 1.13
 
 require (
 	github.com/ahmetb/gen-crd-api-reference-docs v0.1.5
-	github.com/gardener/gardener v1.2.1-0.20200408030154-40b97d31d7f7
-	github.com/gardener/gardener-extensions v1.6.1
+	github.com/gardener/gardener v1.6.0
```
Comment (zanetworker) [inline go.mod:7]: maybe we can vendor latest? 

```suggestion
	github.com/gardener/gardener v1.6.5
```
